presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkTaskInputs.java

```diff
@@ -31,15 +32,22 @@
     private final Map<String, Broadcast<?>> broadcastInputs;
     // For the COORDINATOR_ONLY fragment we first collect the inputs on the Driver
     private final Map<String, List<PrestoSparkSerializedPage>> inMemoryInputs;
+    // fragmentId -> PrestoSparkShuffleReadDescriptor for native execution
+    private final Map<String, PrestoSparkShuffleReadDescriptor> shuffleReadDescriptors;
+    private final Optional<PrestoSparkShuffleWriteDescriptor> shuffleWriteDescriptor;
```

Comment:
These two fields are only used in native execution mode while the first two fields are used in normal Java execution mode. So, some fields are just going to be empty in any mode. Wondering if we can refactor this class into an interface/abstract class and have two implementations 1) PrestoSparkNativeExecutionTaskInputs 2) PrestoSparkJavaExecutionTaskInput  and then you can create the appropriate TaskInput instance inside TaskProcessor.process(). That might just make the logic bit cleaner.  WDYT?